src/internal_rep.rs

```diff
@@ -2320,11 +2320,36 @@ impl ValidatedCall {
             )));
         }
 
-        // Each argument must match the type the function signature expects
-        let mut args = match &call.class_name {
-            Some(c) => vec![convert_class_name_if_this(c, parent_type)?.get_cil_name()],
-            None => Vec::new(),
-        };
+        let mut args =
+            match &call.parent_name {
+                Some(parent_name) => {
+                    match function_info.class {
+                        Some(class_info) => {
+                            if !class_info.inherits.contains(parent_name) {
+                                return Err(CascadeErrors::from(CompileError::new(
+                                    "Invalid Parent",
+                                    file,
+                                    parent_name.get_range(),
+                                    "Resource does not inherit from given parent",
+                                )));
+                            }
+                        }
+                        None => {
+                            return Err(CascadeErrors::from(ErrorItem::Compile(
+                                CompileError::new("No such class", file, call.get_name_range(), ""),
+                            )));
+                        }
+                    };
+                    vec![convert_class_name_if_this(parent_name, parent_type)?.get_cil_name()]
+                }
+                None => {
+                    // Each argument must match the type the function signature expects
```

Comment:
I'm not really sure what the history of this comment is, but it sure seems irrelevant buried down in this block.  I'm not sure it's much better moved outside the block.  It can probably just be deleted, while you're rearranging things here.